src/main/java/com/flair/bi/service/impl/RealmService.java

```diff
@@ -18,20 +30,34 @@
 
     private final RealmRepository realmRepository;
     private final RealmMapper realmMapper;
+    private final UserRepository userRepository;
+    private final UserGroupService userGroupService;
+    private final PasswordEncoder passwordEncoder;
+
 
     @Transactional
     public RealmDTO save(RealmDTO realmDTO) {
         log.debug("Saving realm : {}", realmDTO);
         Realm realm = realmMapper.fromDTO(realmDTO);
         realm = realmRepository.save(realm);
+        userRepository.save(createUser(realm));
```

Comment:
could you elaborate what's the idea here? Should realm service create a user as well? I think realm service should just create a realm.